### PR TITLE
Updated Optuna example

### PR DESCRIPTION
**Pull request by SiddhantSadangi**
# Description

Updated Optuna example code

__Related to:__ Update Optuna example

__Any expected test failures?__
- Notebook tests will fail on Windows and Mac as the kernel needs to be restarted to import lightgbm
- Script tests will fail on Mac due to issue with LightGBM installation on MacOS

---

Add a `[X]` to relevant checklist items

## ❔ This change

- [X] adds a new feature
- [ ] fixes breaking code
- [X] is cosmetic (refactoring/reformatting)

---

## ✔️ Pre-merge checklist

- [X] Refactored code ([sourcery](https://sourcery.ai/))
- [X] Tested code locally
- [X] Precommit installed and run before pushing changes
- [ ] Added code to GitHub tests ([notebooks](workflows/test-notebooks.yml), [scripts](workflows/test-scripts.yml))
- [X] Updated GitHub [README](../README.md)
- [ ] Updated the projects overview page on Notion

---

## 🧪 Test Configuration

- OS: Windows11
- Python version: 3.10.11
- Neptune version: 1.10.4
- Affected libraries with version: neptune-optuna==1.2.0

### CODE REVIEW

File: integrations-and-supported-tools/optuna/notebooks/Neptune_Optuna_integration.ipynb [hunk 1]
```diff
@@ -235,7 +244,9 @@
     ]
    },
    "source": [
-    "### Initialize the NeptuneCallback"
+    "### Initialize the NeptuneCallback\n",
+    "\n",
+    "By default, the callback logs all the plots from the `optuna.visualization` module, details of all trials, and the Study object itself. For how to customize the NeptuneCallback further, see the [docs](integrations/optuna/#customizing-which-plots-to-log-and-how-often)."
```
Comment (normandy7) [inline integrations-and-supported-tools/optuna/notebooks/Neptune_Optuna_integration.ipynb:249]:
```suggestion
    "By default, the callback logs all the plots from the `optuna.visualization` module, details of all trials, and the Study object itself. For how to customize the NeptuneCallback further, see the [integration guide](https://docs.neptune.ai/integrations/optuna/#customizing-which-plots-to-log-and-how-often) in the Neptune docs."
```

File: integrations-and-supported-tools/optuna/notebooks/Neptune_Optuna_integration.ipynb [hunk 2]
```diff
@@ -375,8 +350,8 @@
    "source": [
     "### Log charts and study object after the sweep\n",
     "\n",
-    "If you want to log study metadata after the Study was finished you can use the `.log_study_metadata()`.\n",
-    "`.log_study_metadata()` function logs the same things that `NeptuneCallback` logs, and you can customize what is logged with similar flags."
+    "If you want to log study metadata after the Study is finished you can use the `.log_study_metadata()`.  \n",
+    "This method is generally faster than using `NeptuneCallback`, as it doesn't log the data live. It logs the same metadata as the callback, and acdepts the same flags for customization."
```
Comment (normandy7) [inline integrations-and-supported-tools/optuna/notebooks/Neptune_Optuna_integration.ipynb:354]:
```suggestion
    "This method is generally faster than using `NeptuneCallback`, as it doesn't log the data live. It logs the same metadata as the callback and accepts the same flags for customization."
```

File: integrations-and-supported-tools/optuna/notebooks/Neptune_Optuna_integration.ipynb [hunk 3]
```diff
@@ -637,16 +579,16 @@
...
-    "To compare sweeps between each other or find your current sweep, use Group by:\n",
-    "* Go to the Runs Table\n",
-    "* Click **+ Group by** in the top right\n",
-    "* Type 'sweep-id' and click on it\n",
-    "* Click **Show all** to see your trials in a separate Table View"
+    "To compare trials within each study, use Group by:\n",
+    "1. Go to the Runs Table\n",
+    "1. Click **+ Group by** in the top right\n",
+    "1. Type 'study/study_name' and click on it\n",
+    "1. Click **Show all** to see your trials in a separate Table View"
```
Comment (normandy7) [inline integrations-and-supported-tools/optuna/notebooks/Neptune_Optuna_integration.ipynb:591]:
```suggestion
    "To compare trials within each study, group the runs by study name:\n",
    "1. Go to the runs table.\n",
    "1. Next to the search input box, click **Group by**.\n",
    "1. Type 'study/study_name' and select the field from the list.\n",
    "1. To see your trials in a separate table view, click **Show all**."
```

File: integrations-and-supported-tools/optuna/notebooks/Neptune_Optuna_integration.ipynb [hunk 4]
```diff
@@ -375,8 +350,8 @@
    "source": [
     "### Log charts and study object after the sweep\n",
     "\n",
-    "If you want to log study metadata after the Study was finished you can use the `.log_study_metadata()`.\n",
-    "`.log_study_metadata()` function logs the same things that `NeptuneCallback` logs, and you can customize what is logged with similar flags."
+    "If you want to log study metadata after the Study is finished you can use the `.log_study_metadata()`.  \n",
```
Comment (normandy7) [inline integrations-and-supported-tools/optuna/notebooks/Neptune_Optuna_integration.ipynb:353]:
```suggestion
    "To log study metadata after the Study is finished, use the `log_study_metadata()` function.\n",
```

File: integrations-and-supported-tools/optuna/notebooks/Neptune_Optuna_integration.ipynb [hunk 5]
```diff
@@ -637,16 +579,16 @@
     "## Go to the Neptune app to see your parameter sweep\n",
     "\n",
     "Now when you go to the Neptune app, you have:\n",
-    "* all the trial-level Runs logged with `\"sys/tags\"=\"trial-level\"`\n",
-    "* study-level run logged with `\"sys/tags\"=\"study-level\"`\n",
+    "* all the trial-level Runs logged with `\"sys/tags\"=\"trial\"`\n",
+    "* study-level run logged with `\"sys/tags\"=\"study\"`\n",
```
Comment (normandy7) [inline integrations-and-supported-tools/optuna/notebooks/Neptune_Optuna_integration.ipynb:583]: This is probably sufficient?

```suggestion
    "* all the trial-level runs tagged as `trial`\n",
    "* the study-level run tagged as `study`\n",
```

**Review by normandy7 — CHANGES_REQUESTED** [review]

File: integrations-and-supported-tools/optuna/notebooks/Neptune_Optuna_integration.ipynb [hunk 6]
```diff
@@ -325,43 +336,7 @@
...
-    "    log_plot_contour=False,  # do not create/log plot_contour\n",
-    ")\n",
-    "\n",
-    "# Pass NeptuneCallback to Optuna Study .optimize()\n",
-    "study = optuna.create_study(direction=\"maximize\")\n",
-    "study.optimize(objective, n_trials=5, callbacks=[neptune_callback])\n",
-    "\n",
-    "# Stop logging to a Neptune run\n",
-    "run.stop()"
+    "## More Options"
```
Comment (szaganek) [inline integrations-and-supported-tools/optuna/notebooks/Neptune_Optuna_integration.ipynb:339]:
```suggestion
    "## More options"
```

**Pull request merged**